src/distances/mod.rs

```diff
@@ -92,19 +77,18 @@ pub fn self_dists_all<'a>(
             let start_dist_idx = chunk_idx * CHUNK_SIZE;
             let mut i = calc_row_idx(start_dist_idx, n);
             let mut j = calc_col_idx(start_dist_idx, i, n);
+
             for dist_idx in 0..CHUNK_SIZE {
                 if let Some((k_idx, k_f64)) = k_vals {
-                    let mut j_index = jaccard_index(
+                    let c1 = completeness_vec.map(|cv| cv[i]);
+                    let c2 = completeness_vec.map(|cv| cv[j]);
```

Comment:
I had to look up what `.map()` does here -- might be helpful to note in a comment